### PR TITLE
Hotfix/bcrypt#99

### LINKED ISSUE #99: Switch bcrypt package to node-bcrypt package
**Issue**
The bcrypt package often throws errors causing the app to crash.

**Solution**
Switching to a node compatible version of bcrypt will fix this issue.

### PR DESCRIPTION
**Pull request by DonSeannelly**
This PR moves us to a JavaScript port of bcrypt to *hopefully* avoid the issues we have been facing with it crashing deployments. A test has also been added to check that bcrypt is working so that this will be checked in each build.

Several other issues have been addressed across several files, including improper use of types, bad references, and broken API calls.

Needs Cyber4All/clark-client#474
Resolves #99

### CODE REVIEW

File: src/interactors/user.interactor.spec.ts [hunk 1]
```diff
@@ -4,162 +4,112 @@ import MockDriver from '../drivers/MockDriver';
...
       })
-      .catch(error => {
+      .catch((error) => {
         expect.fail();
         done();
       });
   });
-  it('should return an error message', done => {
+  it('should return an error message', (done) => {
     // Here we are passing an incorrect parameter for DataStore
```
Comment (nvisal1) [inline src/interactors/user.interactor.spec.ts]: We should remove this test. It tests that the function throws an error when datastore is null.

File: src/interactors/user.interactor.spec.ts [hunk 2]
```diff
@@ -4,162 +4,112 @@ import MockDriver from '../drivers/MockDriver';
...
-        done();
-      })
-      .catch(error => {
+      .catch((error) => {
         expect.fail();
         done();
       });
   });
-  it('should return a user. accessGroups should be gone when returned!', done => {
+  it('should return a user. accessGroups should be gone when returned!', (done) => {
```
Comment (nvisal1) [inline src/interactors/user.interactor.spec.ts]: This test might need to be updated. I can't think of a case where we would need to strip off the access groups.

Comment (DonSeannelly) [reply src/interactors/user.interactor.spec.ts]: Yeah, I'm actually not sure about why this test is here. @gshaw1997 do you have any context on this?

Comment (nvisal1) [reply src/interactors/user.interactor.spec.ts]: I wrote the test over the summer. At the time, we were stripping the access group property. I don't think we are/need to anymore though.

Comment (DonSeannelly) [reply src/interactors/user.interactor.spec.ts]: Ah ok. Should this be updated somehow, or just removed?

Comment (nvisal1) [reply src/interactors/user.interactor.spec.ts]: I think removing it would be fine

**Pull request merged**